### DISCUSSION TITLE
How does the API push the pattern into the models?

### DISCUSSION THREAD

**Discussion by JulesCyb** [danielmiessler/Fabric#1154] [Q&A]
Hi, I am wondering how the patterns push the instructions into the models. Lets take chatgpt as an example. Is it like opening a chat window and giving the model it as instructions or is it via playground where it is "system instructions" ?

I am asking because I am unsure how the model reacts, when I create patterns myself. If it is pushed via system instructions how do I set the model configuration such as "Temperature, max tokens, and so on"

**Comment by mattjoyce** [comment]
The pattern and user input is combined and becomes the system prompt.
It's via an api.

you can adjust temperate and maxp with these parameters.
  -t, --temperature=        Set temperature (default: 0.7)
  -T, --topp=               Set top P (default: 0.9)
